repository/src/main/java/org/apache/atlas/repository/store/graph/v2/ClassificationAssociator.java

```diff
@@ -236,6 +242,49 @@ public void setClassifications(Map<String, AtlasEntityHeader> map) throws AtlasB
             RequestContext.get().setDelayTagNotifications(false);
         }
 
+        private Map<String, List<AtlasClassification>> validateAndTransfer(AtlasEntityHeader incomingEntityHeader, AtlasEntityHeader entityToBeChanged) throws AtlasBaseException {
+            Map<String, List<AtlasClassification>> operationListMap = new HashMap<>();
+
+            Set<String> requiredClassificationKeys = Stream.concat(
+                            Optional.ofNullable(incomingEntityHeader.getRemoveClassifications()).orElse(Collections.emptyList()).stream(),
+                            Optional.ofNullable(incomingEntityHeader.getUpdateClassifications()).orElse(Collections.emptyList()).stream()
+                    ).filter(classification -> classification.getEntityGuid().equals(entityToBeChanged.getGuid()))
+                    .map(this::generateClassificationComparisonKey)
+                    .collect(Collectors.toSet());
+
+            Set<String> preExistingClassificationKeys = Optional.ofNullable(entityToBeChanged.getClassifications())
+                    .orElse(Collections.emptyList())
+                    .stream()
+                    .filter(classification -> classification.getEntityGuid().equals(entityToBeChanged.getGuid()))
+                    .map(this::generateClassificationComparisonKey)
+                    .collect(Collectors.toSet());
+
+            Set<String> diff = requiredClassificationKeys.stream()
+                    .filter(key -> !preExistingClassificationKeys.contains(key))
+                    .collect(Collectors.toSet());
+
+            if (!diff.isEmpty()) {
+                String firstTypeName = diff.iterator().next().split("\\|")[1];
+                throw new AtlasBaseException(AtlasErrorCode.CLASSIFICATION_NOT_ASSOCIATED_WITH_ENTITY, firstTypeName);
```

Comment:
Consider collecting all missing classifications and log all classifications

Reply:
@nikhilbonte21 Should we send all the issue inducing classifications in 1 go in the error? The usual error thrown by this method's sibling method is 1 classification per request.